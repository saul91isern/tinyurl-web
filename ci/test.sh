#!/bin/sh

echo "----- Install dependencies -----"
yarn
echo "----- Linter -----"
yarn test --ci

