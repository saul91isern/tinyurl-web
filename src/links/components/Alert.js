import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Message } from "semantic-ui-react";
import { dismissAlert } from "../routines";

export const Alert = ({
  message: { header, info, success, error, text },
  dismissAlert,
}) =>
  text || header ? (
    <Message
      icon={
        (error && "warning sign") || (success && "check") || (info && "info")
      }
      error={error}
      header={header}
      content={text}
      positive={success}
      warning={info}
      onDismiss={dismissAlert}
      visible
    />
  ) : null;

Alert.propTypes = {
  message: PropTypes.object,
  dismissAlert: PropTypes.func,
};

const mapStateToProps = ({ message }) => ({ message });

export default connect(mapStateToProps, { dismissAlert })(Alert);
