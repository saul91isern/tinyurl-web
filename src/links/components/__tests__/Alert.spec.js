/* eslint-disable quotes */
import React from "react";
import { waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { render } from "../../../test/render";
import { Alert } from "../Alert";

describe("<Alert />", () => {
  const defaultMessage = { text: "foo", header: "bar", succes: true };
  const dismissAlert = jest.fn();
  it("renders success message", async () => {
    const { container, queryByText } = render(
      <Alert message={defaultMessage} dismissAlert={dismissAlert} />,
      {}
    );
    expect(queryByText(/foo/)).toBeTruthy();
    expect(queryByText(/bar/)).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it("renders error message", async () => {
    const message = { text: "baz", header: "xyz", error: true };
    const { container, queryByText } = render(
      <Alert message={message} dismissAlert={dismissAlert} />,
      {}
    );
    expect(queryByText(/baz/)).toBeTruthy();
    expect(queryByText(/xyz/)).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it("calls dismissAlert onDismiss", async () => {
    const { container } = render(
      <Alert message={defaultMessage} dismissAlert={dismissAlert} />,
      {}
    );
    userEvent.click(container.querySelector('[class="close icon"]'));
    await waitFor(() => expect(dismissAlert).toHaveBeenCalledTimes(1));
  });
});
