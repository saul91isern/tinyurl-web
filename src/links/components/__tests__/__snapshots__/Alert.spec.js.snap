// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> renders error message 1`] = `
<div>
  <div
    class="ui error icon visible message"
  >
    <i
      aria-hidden="true"
      class="close icon"
    />
    <i
      aria-hidden="true"
      class="warning sign icon"
    />
    <div
      class="content"
    >
      <div
        class="header"
      >
        xyz
      </div>
      <p>
        baz
      </p>
    </div>
  </div>
</div>
`;

exports[`<Alert /> renders success message 1`] = `
<div>
  <div
    class="ui visible message"
  >
    <i
      aria-hidden="true"
      class="close icon"
    />
    <div
      class="content"
    >
      <div
        class="header"
      >
        bar
      </div>
      <p>
        foo
      </p>
    </div>
  </div>
</div>
`;
